app/renderer/abr-document.js

```diff
@@ -257,6 +262,10 @@ AbrDocument.prototype = {
         this.latestGeneration = this.getGeneration();
     },
 
+    setDirty: function () {
+        this.latestGeneration--;
```

Comment:
I think this can lead to unexpected behaviors, especially when going back in the undo stack. I would rather set `this.latestGeneration` to `-1` in order to be sure its value will never be the equal to CodeMirror's `changeGeneration()`.